astropy/units/format/cds.py

```diff
@@ -328,7 +328,9 @@ def _format_superscript(cls, number):
     @classmethod
     def to_string(cls, unit, fraction=False):
```

Comment:
Is `to_string` in any hot-paths? I presume constructing a lambda func here would be a slowdown.

Reply:
I have not checked how this would affect performance.

Reply:
let's add the benchmark label maybe ?

Reply:
Setup:
```shell
$ ipython -i -c 'from astropy import units as u; unit = u.km / u.s**2'
```
On `main`:
```python
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.15.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %timeit u.m.to_string("cds")
2.65 µs ± 6.06 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [2]: %timeit unit.to_string("cds")
21.1 µs ± 132 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```
On this branch:
```python
In [1]: %timeit u.m.to_string("cds")
2.73 µs ± 6 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [2]: %timeit unit.to_string("cds")
20.7 µs ± 118 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```
There does not seem to be any noteworthy performance change.

Reply:
FWIW I added the `benchmark` label. Does not hurt to have extra data points. Thanks!

Reply:
Even if the `lambda` is causing a performance regression then it will still only affect the `"cds"` format, and the benchmarks don't use that at all.

Reply:
If it's not slowing us down...